index.js

```diff
@@ -53,18 +58,52 @@ LiveReloadPlugin.prototype.start = function start(watching, cb) {
 };
 
 LiveReloadPlugin.prototype.done = function done(stats) {
+  var timestamps = stats.compilation ? stats.compilation.fileTimestamps : {};
+  var assets = stats.compilation.assets;
+  var include = [];
+
+  // If no timestamp information is available, use all assets.
+  if (!timestamps) {
+    include.push.apply(include, Object.keys(assets));
+  }
+  else {
+    this.changedFiles = Object.keys(timestamps).filter(function(watchfile) {
+      return this.startTime < Math.ceil(statSync(watchfile).mtime);
```

Comment:
Synchronously `stat`ing all files seems like it could be a cause for a hang in a larger product.

Reply:
True, and I suspect webpack has already stat'd these files at some other point. Wonder if there is a way to tap into a stat cache of some kind?

Reply:
In the webpack wiki you linked, it looks like they're using prevTimestamps as a way to see changes.

```js
 var changedFiles = Object.keys(compilation.fileTimestamps).filter(function(watchfile) {
      return (this.prevTimestamps[watchfile] || this.startTime) < (compilation.fileTimestamps[watchfile] || Infinity);
    }.bind(this));
```